source/common/compressor/compressor_impl.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "envoy/compressor/compressor.h"
+
+#include "zlib.h"
+
+namespace Envoy {
+namespace Compressor {
+
+/**
+ * Implementation of compressor's interface.
+ */
+class CompressorImpl : public Compressor {
```

Comment:
nit: Would prefer to call this `ZlibCompresorImpl` since we may have other compression types in the future.